### PR TITLE
chore: Add Django 5.0 and Python 3.12 to the testing

### PR DESCRIPTION
**Pull request by cclauss**
# I'm helping!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc.)
- [x] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Add Django 5.0 and Python 3.12 to the testing

### CODE REVIEW

**Comment by Dresdn** [comment]
Thank you @cclauss!

**Comment by Dresdn** [comment]
Downstream PyYAML needs to be upgraded to `6.0.1` to support Python 3.12.

**Comment by cclauss** [comment]
Yes, and https://github.com/yaml/pyyaml/pull/769 is still open but Py3.12 wheels are available.  https://pypi.org/project/PyYAML/6.0.1/#files

Let's run the tests on pyyaml 6.0.1 and if that fails, we can hold off on the Py3.12 upgrade and just focus on Dj5.

File: pyproject.toml [hunk 1]
```diff
@@ -70,7 +72,9 @@ build_command = "pip install poetry && poetry build"
 
 [tool.poetry.dependencies]
 python = "^3.8"
-django = ">=3.2,<4.3"
+django = ">=3.2,<5.1"
+pyyaml = { version = "^6.0.1", python = "^3.12" }
+setuptools = { version = "^69.0.3", python = "^3.12" }
```
Comment (Dresdn) [inline pyproject.toml]: Shouldn't need the `setuptools` declaration.

File: pyproject.toml [hunk 2]
```diff
@@ -70,7 +72,9 @@ build_command = "pip install poetry && poetry build"
 
 [tool.poetry.dependencies]
 python = "^3.8"
```
Comment (Dresdn) [inline pyproject.toml:74]: Looking at it, `wemake-python-styleguide` should be upgraded to `0.18.0` as it'll pull in the updated version of bandit once released, but needs python to be `3.8.1`.

Comment (cclauss) [reply pyproject.toml:74]: Not using ruff yet?  https://docs.astral.sh/ruff

File: pyproject.toml (hunk 1 above)
Comment (cclauss) [inline pyproject.toml]:
```suggestion
```

File: pyproject.toml (hunk 2 above)
Comment (Dresdn) [reply pyproject.toml:74]: Funny you say that, as that's 100% on my ToDo. I started a working branch a while ago, but yes, we absolutely should.

**Comment by cclauss** [comment]
Please let me know if you want me to squash on my side or if you just `Squash & Merge`.

**Comment by Dresdn** [comment]
I'll squash it - thank you for the help!

**Pull request merged**